vllm/v1/core/sched/scheduler.py

```diff
@@ -213,6 +214,76 @@ def __init__(
         self.use_pp = self.parallel_config.pipeline_parallel_size > 1
         self.use_v2_model_runner = envs.VLLM_USE_V2_MODEL_RUNNER
 
+        def has_mamba_layers(kv_cache_config: KVCacheConfig) -> bool:
+            has_mamba: bool = any(
+                isinstance(group_spec.kv_cache_spec, MambaSpec)
+                for group_spec in kv_cache_config.kv_cache_groups
+            )
+            if vllm_config.model_config.is_hybrid:
+                assert has_mamba, "Hybrid models must have mamba layers"
+            return has_mamba
+
+        self.need_mamba_block_aligned_split = (
+            has_mamba_layers(self.kv_cache_config)
+            and self.cache_config.mamba_cache_mode == "align"
+        )
+
+    def _mamba_block_aligned_split(
+        self,
+        request: Request,
+        num_new_tokens: int,
+        num_new_local_computed_tokens: int = 0,
+        num_external_computed_tokens: int = 0,
+    ) -> int:
+        assert num_external_computed_tokens == 0, (
+            "External KV connector is not verified yet"
+        )
+        if request.num_output_tokens == 0:  # prefill
+            # Ensure new tokens for a request in the prefill phase do not contain
+            # sps tokens, especially in the last prefill chunk. For a hybrid-model,
+            # extra sps tokens would corrupt the generated Mamba state.
+            # TODO: This logic does not yet handle resumed requests.
+            if request.num_computed_tokens < request.num_prompt_tokens:
```

Comment:
I've moved this adjustment here.
Can you help to
1. add more comment to explain the problem
2. verify resumed requests